src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -307,6 +333,12 @@ private bool TryReplaceBannedFunction(FunctionExpression expression, [NotNullWhe
             return false;
         }
 
+        private bool IsCaseInsensitiveCondition(FunctionExpression left, FunctionExpression right)
+        {
+            return (left.Function == "toLower" && right.Function == "toLower")
+                || (left.Function == "toUpper" && right.Function == "toUpper");
```

Comment:
>toUpper [](http://example.com/codeflow?start=69&length=7)

Only the function that matches how we compile the operator exactly should be decompiled into the case-insensitive operator.

The choice we made between `ToUpper()` and `ToLower()` for the case-insensitive operators wasn't arbitrary. Some languages (Turkish, for example) have multiple lowercase letters that map to a single uppercase letter. If we decompile `ToLower()` usage into the operator, we would change the meaning slightly in those cases.